src/Files.App/Filesystem/FilesystemOperations/Helpers/FilesystemHelpers.cs

```diff
@@ -538,6 +548,16 @@ public async Task<ReturnResult> RenameAsync(IStorageItemWithPath source, string
 			var progress = new Progress<FileSystemProgress>();
 			progress.ProgressChanged += (s, e) => returnStatus = returnStatus < ReturnResult.Failed ? e.Status!.Value.ToStatus() : returnStatus;
 
+			if (string.IsNullOrWhiteSpace(newName)
+				|| ContainsRestrictedCharacters(newName)
+				|| ContainsRestrictedFileName(newName))
+			{
+				await DialogDisplayHelper.ShowDialogAsync(
+					"ErrorDialogThisActionCannotBeDone".GetLocalizedResource(),
+					"ErrorDialogNameNotAllowed".GetLocalizedResource());
+				return ReturnResult.Failed;
```

Comment:
This block is almost the same as l.103 to l.107 aside of the return time. Would it be possible to fuse them? I'm not sure how, it's not huge. Except if you find an idea right away, do not spend too much on it.

Reply:
I don't think there is an easy way to merge them without modifying changing a lot of code

Reply:
Agreeing with you on this one, let's not overdo it.